src/core/settings/settings.cpp

```diff
@@ -0,0 +1,174 @@
+ *
+ * $END_LICENSE$
+*/
+
+#include "settings.h"
+
+#include <QDir>
+#include <QDebug>
+#include <QJsonDocument>
+#include <QJsonObject>
+
+Settings::Settings(QObject *parent) : QObject(parent)
+{
+    // Create exposed configuration objects
+    startConfigChanged(m_startConfig = new StartConfig(this));
+    searchConfigChanged(m_searchConfig = new SearchConfig(this));
+
+    // Check for settings directory
+    if (!QDir(APP_CONFIG_LOCATION).exists()) {
+        qDebug() << "Settings path doesn't exist.";
+        qDebug() << "Creating" << APP_CONFIG_LOCATION << "...";
+        QDir().mkpath(APP_CONFIG_LOCATION);
+    }
+    // Check for settings file
+    if (!QFile(SETTINGS_FILENAME).exists()) {
+        qDebug() << "Settings file doesn't exist.";
+        qDebug() << "Creating" << SETTINGS_FILENAME << "...";
+        QFile file(SETTINGS_FILENAME);
+        if (!file.open(QIODevice::WriteOnly)) {
+            qWarning("Couldn't open settings file for write!");
+        }
+        else {
+            QTextStream stream(&file);
+            stream << defaultJSON();
+            qDebug() << "Default settings file created.";
+        }
+    }
+}
+
+void Settings::load()
+{
+    QFile file(SETTINGS_FILENAME);
+    if (!file.open(QIODevice::ReadOnly)) {
+        qWarning("Couldn't open settings file for read!");
+        return;
+    }
+
+    QByteArray bytes = file.readAll();
+    QJsonDocument doc(QJsonDocument::fromJson(bytes));
+
+    QJsonObject root = doc.object();
+    QJsonObject meta = root["meta"].toObject();
+    QString metaSchema = meta["schema"].toString();
+    if (metaSchema != "0.1") {
+        qWarning() << "Unknown settings schema version " << metaSchema  << "!";
+        return;
+    }
+    QJsonObject data = root["data"].toObject();
+    QJsonObject dataStart = data["start"].toObject();
+    m_startConfig->setStartUrl(QUrl(dataStart["url"].toString()));
+    QJsonObject dataSearch = data["search"].toObject();
+    QString searchEngineString = dataSearch["engine"].toString();
+    SearchConfig::SearchEngine searchEngine;
+    if (searchEngineString == "duckduckgo")
+        searchEngine = SearchConfig::SearchEngine::DuckDuckGo;
+    else if (searchEngineString == "google")
+        searchEngine = SearchConfig::SearchEngine::Google;
+    else if (searchEngineString == "bing")
+        searchEngine = SearchConfig::SearchEngine::Bing;
+    else if (searchEngineString == "yahoo")
+        searchEngine = SearchConfig::SearchEngine::Yahoo;
+    else
+        searchEngine = SearchConfig::SearchEngine::Custom;
+    m_searchConfig->setSearchEngine(searchEngine);
+    m_searchConfig->setCustomSearchUrl(QUrl(dataSearch["custom_url"].toString()));
+    setDirty(false);
+}
+
+void Settings::save()
+{
+    QFile file(SETTINGS_FILENAME);
+    if (!file.open(QIODevice::WriteOnly)) {
+        qWarning("Couldn't open settings file for write!");
+        return;
+    }
+    QTextStream stream(&file);
+    stream << json();
+    setDirty(false);
```

Comment:
Done in https://github.com/lirios/browser/pull/17/commits/c1158874f5a04893269e4a553528ee58ab2ebd46